plugin/trino-base-jdbc/src/main/java/io/trino/plugin/jdbc/JdbcPageSource.java

```diff
@@ -133,7 +135,22 @@ else if (javaType == Slice.class) {
             pageBuilder = new PageBuilder(columnHandles.stream()
                     .map(JdbcColumnHandle::getColumnType)
                     .collect(toImmutableList()));
-            resultSetFuture = UNINITIALIZED_RESULT_SET_FUTURE;
+            resultSetFuture = supplyAsync(() -> {
+                long start = nanoTime();
+                try {
+                    log.debug("Executing: %s", statement);
+                    return statement.executeQuery();
+                }
+                catch (SQLException e) {
+                    throw handleSqlException(e);
+                }
+                finally {
+                    readTimeNanos.addAndGet(nanoTime() - start);
+                }
+            }, executor).thenAccept(resultSet -> {
+                this.resultSet = requireNonNull(resultSet, "resultSet is null");
+                buildPageFromResultSet();
```

Comment:
Which thread does this work? If it's not the driver thread, then this means the connector is "cheating" and using more CPU than it's allocated by the engine.

Reply:
It uses the executor that installed in the `JdbcModule`, I think it is not the driver thread.

Checked the executor was introduced for preparing `resultSet` in a async way, is it a way also "cheating"
cc @raunaqmorarka 

Would it make sense to run `buildPageFromResultSet` on the current thread instead of using the thread pool? @martint

Reply:
Thanks, switch consuming resultSet part to use main thread.